src/dev/build/lib/archive_utils.ts

```diff
@@ -0,0 +1,93 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+import fs from 'fs';
+import yauzl from 'yauzl';
+import yazl from 'yazl';
+
+// The utilities in this file are largely copied with minor modifications from
```

Comment:
Can we look into using re-using the existing fs utils in `src/dev/build/lib/fs.ts`? 

I'm going to remove yauzl/yazl in favor of `archiver` in the near future, the former hasn't been updated in 4 years and we can consolidate.  If it's still needed for now that's fine, just providing context.

Reply:
I took a look and this seems like a slightly larger undertaking than I expected, just based on how Fleet expects to parse package manifests, etc. I think it'd be very valuable to consolidate though, especially with the lack of maintenance on our current archive tools. I filed https://github.com/elastic/kibana/issues/139840 to migrate this bundling task and Fleet's other archive logic to `archiver` and `zlib` which appear to be more maintained solutions.